### ISSUE TITLE
Currency inconsistencies between account and budget

### ISSUE THREAD

**Issue by Limezy** [firefly-iii/firefly-iii#1344]
I am running Firefly III version 4.7.2.2

I have some troubles using two currencies in my setup.
I'm French so I want my base currency to be €. 
I'm living in China so I want some accounts and budgets to be in Chinese RMB.

I found some currency inconsistencies in the software, here is one example : 
I create two asset accounts, let's say one is "FR" in €, other is "CN" in RMB.
I create a budget "B". My base currency is € so I have no choice, my budget shows in €. Fair enough.

Let's say for example I make 500 RMB expense from "CN" that belong to the "B" budget.
I create a new withdrawal from "CN" and link it to "B" with the amount "500 RMB"
Saddly when I go to check the "B" budget, it shows "Spent : -500€" !

I'm afraid this bug will prevent me to use Firefly. I could of course do everything in € but this would make my accounting very complicated as I would have to make the conversion manually for each of my inputs !

Would it be possible to add an automatic conversion from one currency to the base currency ? Even a fixed rate would make me happy ! (I have a fixed rate in my working contract, and anyway I could make some "corrective" transactions from time to time)

Thanks for your help !

**Comment by JC5** [comment]
Multi-currency transactions are possible already. Just change the currency of a transaction on your RMB account to EUR and the options will appear. Firefly III can even do the currency conversion for you.

A solution for multi-currency budgeting is forthcoming. I apologize if the tool is unusable for you until then.

**Comment by Limezy** [comment]
I was not aware of that !

It looks usable as a workaround solution.
1. I create a new transaction from my chinese account but convert it directly in €
(Let's say 70 RMB = 10 €)
2. I associate it with my budget that is already in €
The only problem now is a cosmetic one : in my budget I see that I have spent 10 RMB

I will try to use Firefly as it is, waiting for a more consistent multi-currency support !

Of course you don't have to appologize for that. You are already giving for free an awsome piece of software and I understand that my "complex" situation is not standard. Believe me I've tried a bunch of personnal accounting softwares (including paying ones) and regarding multi-currency support, a LOT of them failed short way before Firefly.

Nevertheless I'm glad that better multi-currency support is forthcoming and will keep an eye on every updates. If you want to have a better understanding on what us, multi-currencies users need feel free to ask !
